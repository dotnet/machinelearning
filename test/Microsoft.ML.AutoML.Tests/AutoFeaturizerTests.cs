// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json;
using Microsoft.ML.TestFramework;
using Xunit;
using Xunit.Abstractions;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using System.Text.Json.Serialization;

namespace Microsoft.ML.AutoML.Test
{
    public class AutoFeaturizerTests : BaseTestClass
    {
        private readonly JsonSerializerOptions _jsonSerializerOptions;

        public AutoFeaturizerTests(ITestOutputHelper output)
            : base(output)
        {
            _jsonSerializerOptions = new JsonSerializerOptions()
            {
                WriteIndented = true,
                Converters =
                {
                    new JsonStringEnumConverter(), new DoubleToDecimalConverter(), new FloatToDecimalConverter(),
                },
            };

            if (Environment.GetEnvironmentVariable("HELIX_CORRELATION_ID") != null)
            {
                Approvals.UseAssemblyLocationForApprovedFiles();
            }
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("ApprovalTests")]
        public void AutoFeaturizer_uci_adult_test()
        {
            var context = new MLContext(1);
            var dataset = DatasetUtil.GetUciAdultDataView();
            var pipeline = context.Auto().Featurizer(dataset, outputColumnName: "OutputFeature", excludeColumns: new[] { "Label" });

            Approvals.Verify(JsonSerializer.Serialize(pipeline, _jsonSerializerOptions));
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("ApprovalTests")]
        public void AutoFeaturizer_iris_test()
        {
            var context = new MLContext(1);
            var dataset = DatasetUtil.GetIrisDataView();
            var pipeline = context.Auto().Featurizer(dataset, excludeColumns: new[] { "Label" });

            Approvals.Verify(JsonSerializer.Serialize(pipeline, _jsonSerializerOptions));
        }
    }
}
