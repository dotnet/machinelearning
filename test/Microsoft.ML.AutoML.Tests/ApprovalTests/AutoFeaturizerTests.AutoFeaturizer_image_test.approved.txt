{
  "schema": "e0 * e1 * e2 * e3 * e4",
  "currentSchema": "e0 * e1 * e2 * e3 * e4",
  "estimators": {
    "e0": {
      "estimatorType": "LoadImages",
      "parameter": {
        "OutputColumnName": "ImagePath",
        "InputColumnName": "ImagePath"
      }
    },
    "e1": {
      "estimatorType": "ResizeImages",
      "parameter": {
        "OutputColumnName": "ImagePath",
        "InputColumnName": "ImagePath",
        "ImageHeight": 224,
        "ImageWidth": 224,
        "CropAnchor": "Center",
        "Resizing": "Fill"
      }
    },
    "e2": {
      "estimatorType": "ExtractPixels",
      "parameter": {
        "OutputColumnName": "ImagePath",
        "InputColumnName": "ImagePath",
        "ColorsToExtract": "Rgb",
        "OrderOfExtraction": "ARGB",
        "OutputAsFloatArray": true
      }
    },
    "e3": {
      "estimatorType": "DnnFeaturizerImage",
      "parameter": {
        "OutputColumnName": "ImagePath",
        "InputColumnName": "ImagePath",
        "ModelFactory": "resnet_18"
      }
    },
    "e4": {
      "estimatorType": "Concatenate",
      "parameter": {
        "InputColumnNames": [
          "ImagePath"
        ],
        "OutputColumnName": "Features"
      }
    }
  }
}